docs/core/diagnostics/microsoft-diagnostics-netcore-client.md

```diff
@@ -0,0 +1,459 @@
+
+```csharp
+public string Name { get; }
+```
+
+The name of the Provider
+
+```csharp
+public EventLevel EventLevel { get; }
+```
+
+The EventLevel of the given instance of [`EventPipeProvider`](#class-eventpipeprovider).
+
+```csharp
+public long Keywords { get; }
+```
+
+A long that represents bitmask for keywords of the EventSource.
+
+```csharp
+public IDictionary<string, string> Arguments { get; }
+```
+
+An `IDictionary` of key-value pair string representing optional arguments to be passed to EventSource representing the given `EventPipeProvider`.
+
+#### Remarks
+
+This class is immutable, as EventPipe does not allow a provider's configuration to be modified during an EventPipe session (as of .NET Core 3.1).
+
+### class EventPipeSession
+
+```csharp
+public class EventPipeSession : IDisposable
+{
+    public Stream EventStream { get; }
+    public void Stop();
+}
+```
+
+This class represents an ongoing EventPipe session that has been started. It is immutable and acts as a handle to an EventPipe session of the given runtime.
+
+```csharp
+public Stream EventStream { get; }
+```
+
+Returns a `Stream` that can be used to read the event stream.
+
+```csharp
+public void Stop();
+```
+
+Stops the given EventPipe session.
+
+### enum DumpType
+
+```csharp
+public enum DumpType
+{
+    Normal = 1,
+    WithHeap = 2,
+    Triage = 3,
+    Full = 4
+}
+```
+
+Represents the type of dump that can be requested.
+
+* `Normal`: Include just the information necessary to capture stack traces for all existing traces for all existing threads in a process. Limited GC heap memory and information.
+* `WithHeap`: Includes the GC heaps and information necessary to capture stack traces for all existing threads in a process.
+* `Triage`: Include just the information necessary to capture stack traces for all existing traces for all existing threads in a process. Limited GC heap memory and information.
+* `Full`: Include all accessible memory in the process. The raw memory data is included at the end, so that the initial structures can be mapped directly without the raw memory information. This option can result in a very large dump file.
+
+### Exceptions
+
+Exceptions that are thrown from the library are `DiagnosticsClientException`s or a derived class of it.
+
+```csharp
+public class DiagnosticsClientException : Exception
+```
+
+#### UnsupportedProtocolException
+
+```csharp
+public class UnsupportedProtocolException : DiagnosticsClientException
+```
+
+This may be thrown when the command is not supported by either the library or the target process' runtime.
+
+#### ServerNotAvailableException
+
+```csharp
+public class ServerNotAvailableException : DiagnosticsClientException
+```
+
+This may be thrown when the runtime is not available for diagnostics IPC commands, such as early during runtime startup before the runtime is ready for diagnostics commands, or when the runtime is shutting down.
+
+#### ServerErrorException
```

Comment:
Missing `UnsupportedCommandException`. Also, should we document `DiagnosticsClientException` generically as the exceptions that are thrown from the library for it's operations?